x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/whitelist/InternalSqlScriptUtils.java

```diff
@@ -221,14 +221,14 @@ public static Number tan(Number value) {
     //
     // Date/Time functions
     //
-    public static Integer dateTimeChrono(Object dateTime, String tzId, String chronoName) {
-        if (dateTime == null || tzId == null || chronoName == null) {
+    public static Integer dateTimeExtract(Object dateTime, String tzId, String extractorName) {
```

Comment:
I don't have a strong preference regarding the method rename however there is an associated cost for it in backwards compatibility.
When executing across older clusters, the generated query would fail since there's no `dateTimeExtract` in those versions, only `dateTimeChrono`.

To avoid this, simply keep `dateTimeChrono` around and redirect it to `dateTimeExtract` so its backwards compatible or avoid the method rename in the first place.

Reply:
Thanks, good catch. It is not just a simple method rename, also the possible values of the last parameter changed (from `ChronoField` enum to `DateTimeExtract` enum values). I will keep around a `@Deprecated dateTimeChrono()` for backwards compatibility that will do this translation. I guess we can remove it in the next major version.